src/LIBSVM.jl

```diff
@@ -143,6 +142,25 @@ function instances2nodes{U<:Real}(instances::AbstractMatrix{U})
     (nodes, nodeptrs)
 end
 
+function instances2nodes!{U<:Real}(nodes, nodeptrs, instances::AbstractMatrix{U})
+    nfeatures = size(instances, 1)
+    ninstances = size(instances, 2)
+    @assert size(nodes) == (nfeatures+1,ninstances)
+    @assert size(nodeptrs,1) == ninstances
+
+    for i=1:ninstances
+        k = 1
+        for j=1:nfeatures
+            nodes[k, i] = SVMNode(Int32(j), Float64(instances[j, i]))
+            k += 1
+        end
+        nodes[k, i] = SVMNode(Int32(-1), NaN)
+        nodeptrs[i] = pointer(nodes, (i-1)*(nfeatures+1)+1)
+    end
+
+    (nodes, nodeptrs)
+end
```

Comment:
Better not to duplicate code here, and instead make `instances2nodes` call `instances2nodes!`